scripts/apps/packaging/services/PackagesService.js

```diff
@@ -123,6 +123,34 @@ export function PackagesService(api, $q, archiveService, lock, autosave, authori
         };
     };
 
+    this.setItemLabel = function(item, label) {
+        var pkg = authoringWorkspace.getItem();
+
+        _.forEach(pkg.groups, (group) => {
+            var ref = _.find(group.refs, {guid: item._id});
+
+            if (ref) {
+                ref.label = label ? label.qcode : null;
```

Comment:
could we store label as `{qcode, name}` like we do for other cvs ?

Reply:
What if the label name changes? Should I scan the archive to replace the name everywhere?